src/conn.rs

```diff
@@ -14,36 +14,60 @@ use crate::RequestId;
 
 use lber::structures::{Null, Tag};
 
-#[cfg(feature = "tls")]
+#[cfg(any(feature = "tls-native", feature = "tls-rustls"))]
 use futures_util::future::TryFutureExt;
 use futures_util::sink::SinkExt;
-#[cfg(feature = "tls")]
+#[cfg(feature = "tls-native")]
 use native_tls::TlsConnector;
 #[cfg(unix)]
 use percent_encoding::percent_decode;
+#[cfg(feature = "tls-rustls")]
+use rustls::ClientConfig;
 use tokio::io::{self, AsyncRead, AsyncWrite};
 use tokio::net::TcpStream;
 #[cfg(unix)]
 use tokio::net::UnixStream;
 use tokio::stream::StreamExt;
 use tokio::sync::mpsc;
-#[cfg(feature = "tls")]
+#[cfg(any(feature = "tls-native", feature = "tls-rustls"))]
 use tokio::sync::oneshot;
 use tokio::time;
-#[cfg(feature = "tls")]
+#[cfg(all(feature = "tls-native", not(feature = "tls-rustls")))]
 use tokio_native_tls::{TlsConnector as TokioTlsConnector, TlsStream};
+#[cfg(all(feature = "tls-rustls", not(feature = "tls-native")))]
+use tokio_rustls::{client::TlsStream, TlsConnector as TokioTlsConnector};
+#[cfg(not(any(feature = "tls-native", feature = "tls-rustls")))]
+compile_error!(r#"Either feature "tls-native" or "tls-rustlsls" must be enabled for this crate"#);
```

Comment:
It must be possible to compile the crate without _any_ TLS support, so this test is superfluous.